azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/method_call.py

```diff
@@ -0,0 +1,39 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+"""This module contains a class representing method calls that are received and responded to.
+"""
+
+
+class MethodCall(object):
```

Comment:
>MethodCall [](start = 6, length = 10)

on the device side i'd rather call this a MethodRequest (the "call" imho is on the service side). it's something we receive, not something we call.

Reply:
My issue with this is that we would then be passing the MethodRequest back into the client.send_method_response() API, which is bad semantics.

---
In reply to: [266186085](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266186085) [](ancestors = 266186085)

Reply:
technically the only thing we need to pass from the request into send_method_response is the request_id, not the full request object. we have to name this object for what it is, not what it gets passed to. You may go into other SDKs and check what they are doing and how they are naming things, but MethodCall doesn't feel nearly as accurate or used as MethodRequest IMHO.

---
In reply to: [266625788](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266625788) [](ancestors = 266625788,266186085)